README.md

```diff
@@ -188,113 +252,166 @@ The following CSS:
+ - CSS variables may be declared in `:hover` and other psuedo-classes, and they are evaluated properly.
+ - Variables in nested rules can be deduced with the help of [`postcss-nested`](https://github.com/postcss/postcss-nested) or [`postcss-nesting`](https://github.com/jonathantneal/postcss-nesting).
 
+Continue to the next section to see where some of these might be unsafe to do. There are reasons behind the ethos of why the other plugin, `postcss-custom-properties`, is very limited in what it supports, due to differing opinions on whether broader, yet potentially incorrect, support is okay.
 
 
-# Syntax
+# Caveats
 
-### [Defining Custom Properties: the --* family of properties](http://dev.w3.org/csswg/css-variables/#defining-variables)
+When declaring a CSS variable inside one selector, but consume it in another, this does make an unsafe assumption which can be non-conforming in certain edge cases. Here is an example where these limitations unavoidably result in non-conforming behavior.
 
-A custom property is any property whose name starts with two dashes (U+002D HYPHEN-MINUS). A property must be in a `rule`.
+Using the following CSS:
 
-*Note: `:root` is nothing more than the selector for the root DOM node. You could use any other selector (class, id, etc)*
-```
-:root {
-	--foo-width: 100px;
-	--foo-bg-color: rgba(255, 0, 0, 0.85);
+```css
+.component {
+  --text-color: blue;
 }
-```
 
-A custom property can be declared multiple times.
+.component .header {
+  color: var(--text-color);
+}
 
-*See: [W3C Draft: CSS Custom Properties for Cascading Variables, section 2](http://dev.w3.org/csswg/css-variables/#defining-variables) for more information.*
+.component .text {
+  --text-color: green;
+  color: var(--text-color);
+}
+```
+   
+When nesting the markup like this, you may get incorrect behavior, because this only knows the CSS inheritance, not the HTML structure. (Note the innermost `<h1 class="Title">`.)
 
+```html
+<div class="component">
+    Black
 
-### [Using Cascading Variables: the var() notation](http://dev.w3.org/csswg/css-variables/#using-variables)
+    <h1 class="title">
+        Blue
 
+        <p class="decoration">
+            Green
 
+            <h1 class="title">Blue with this plugin, but green per spec</h1>
+        </p>
+    </h1>
+</div>
 ```
-.foo {
-	width: var(--foo-width);
-	/* You can even provide a fallback */
-	background: var(--foo-bg-color, #ff0000);
-}
-```
-
-*See: [W3C Draft: CSS Custom Properties for Cascading Variables, section 3](http://dev.w3.org/csswg/css-variables/#using-variables) for more information.*
 
+This spec deviation is intentional, as there's nothing this tool can do about that. [`postcss-custom-properties`](https://github.com/postcss/postcss-custom-properties) avoids this problem entirely by restricting itself to just the `:root` selector. This is because the developers there would prefer to not support a feature instead of supporting it almost-correctly like what this plugin does.
```

Comment:
I feel "correct" is a little semantically correct than "perfect", since correctness deals with accuracy, while perfection deals with flawlessness.

Are you and the other plugin's dev team aiming for a flawless implementation of the spec, or merely a faithful, accurate one for what is implemented?